### ISSUE TITLE
Bios Mod with New Bios 1.56 Bricks my x1c6 (i5-8350u)

### ISSUE THREAD

**Issue by jaejunh** [tylernguyen/x1c6-hackintosh#172]
I picked up refurbished x1c6 from the store.
Before I installed unmodified BIOS Monterey 12.2.1 with OC 0.8.3, 
I updated  most current BIOS 1.56.  

It was sort of working well except launching Accessibility App failed (as well as Siri app).   So I decided to 'mod' the bios as the site recommended.  

So I decided to follow the BIOS mod option, but as soon as I flashed the patched BIOS, my x1c6 just bricked.

- Symptom:   when I connect power, the usb power port lit up, and it the power button blink 3 times.  That's it.  Nothing happens.

- My failed recovery attempts:  try to restore original_bios.img to Winbond bios chip by:
flashrom -p ch341a_spi -w original_bios.img
==>  that didn't change anything.  Stayed bricked.   

Do you guys have any idea to unbrick my x1c6?

**Comment by tylernguyen** [comment]
@jaejunh 

I just freshly updated my BIOS to 1.56 and modded it just fine. It is not an issue with the BIOS mod.

Try resetting the laptop using the pinhole button close to the miroSD card reader.

**Comment by jaejunh** [comment]
Thank you for your quick reply, and I am glad to know that it should have worked for me (yet it didn't for myself)

As you recommended, I punched in pinhole several times, yet still cannot turn on (no red lenovo log for me)

To get some idea, here is what I did (to patch bios and to brick my x1c6 )

Preparing x1c6:
0) x1c6 was installed with 3 OS before dumping bios:
   EFI partition (nvme0n1p1)
   Monterey 12.2.1  (nvme0n1p2)
   Windows 10 (nvme0n1p3)
   Ubuntu 20.04 (nvme0n1p4)
   
   EFI contains bootloaders:   OC, Windows. Ubuntu(grub), refind
   
1)  Enter Setup > Config > Power > Turn Off > Disable Builtin Battery  -->  That makes my x1c6 turn off for maintenance.  BTW, I didn't pull out the battery and coin ceil battery.

2)  Dump the bios twice via ch134a and clips and diffing two files reveals they are equal --> I conclude dump was OK

3)  Since I didn't want to reclip the bios chip, I let it attached while I am making patched bios

4) To maximizing the patching, kept the *_x80_patch.txt uncommented.  Then, I issued following,

./UEFIPatch orig.bios-1.img xx70_xx80_patches_v7.txt -o bios_patched.img

... Interesting result was....
jaejunh@t490s-linux:~/thinkpad_hack/old$ ./UEFIPatch orig.bios-1.img xx70_xx80_patches_v7.txt -o bios_patched.img
parseVolume: unknown file system FFF12B8D-7696-4C8B-A985-2747075B4F50
**_parseBios: one of volumes inside overlaps the end of data_**
_**parseFile: non-empty pad-file contents will be destroyed after volume modifications**_
patch: replaced 16 bytes at offset 3B60h 04320B483CC2E14ABB16A73FADDA475F -> 778B1D826D24964E8E103467D56AB1BA
patch: replaced 16 bytes at offset 118D0h 04320B483CC2E14ABB16A73FADDA475F -> 778B1D826D24964E8E103467D56AB1BA
patch: replaced 28 bytes at offset 6B83Ah 09072C1300000209072D1300000309072E1300000409072F13000005 -> 09072C1300001909072D1300001A09072E1300001B09072F1300001C
patch: replaced 58 bytes at offset 53B64h 00143100300030004D0068007A000000143100350030004D0068007A000000143200300030004D0068007A000000143200350030004D0068007A -> 00143100320035003000200020000000143100330030003000200020000000143100330035003000200020000000143100340030003000200020
patch: replaced 26 bytes at offset 69E0Dh 0A821286A10B04001206A60B010016021206A60B020016022902 -> 0A821286A10B04001206A60B000016021206A60B000016022902
patch: replaced 11 bytes at offset 545h 0BC8394B040F8437020000 -> 0BC8394B04E93802000000
patch: replaced 5 bytes at offset 535h C8390B7516 -> C8390B7500
Image patched


I didn't like that above _**'warning/error' message**_
but kept going.

5) I hex edited  patched_bios.img with *the string* turning last one to 'FF' created edited_patched_bios.img.   I repeat again with second copy of patched file.  and compared two-->OK
6) Undump/write new edited_patched_bios.img  to bios chip via ch134a again.

7)  By reconnecting AC power,  computer power button blinks 3 times, and nothing happens --> Brick

8) At this time, I was sweeting, and I try to put original  bios 
ch134a , but it didn't changed a thing.  

9) Since 8) didn't work,  reviewing the process,   I created edited_original_bios with "the sting"  (I guessed even original bios may be need to be hex-edited with "the string".  my assumption here was that without the patch/modification,  I still need to hex-edit old one)
--> that didn't change a bit either.
-------------------------------------------------------------

Can you spot any of my mistake?   

Btw, after the bricked status,  I have no way to 'turn off' the computer again via setup menu (since  It's  bricked), so I pull out battery/coin cell and repeat recovery process, and it didn't work either.

**Comment by tylernguyen** [comment]
@jaejunh 

You have to hold the buttom. Pushing it won't doing anything. See https://conetrix.com/blog/lenovo-thinkpad-emergency-reset-hole

**Comment by jaejunh** [comment]
@tylernguyen thank for the follow up.  I will try proper pin hole reset method.  If that doesn't work, I will restore the original  bios and pin hole reset as you guided (though I can do that this weekend)

**Comment by jaejunh** [comment]
@tylernguyen well, it turns out that I did serious physical damage to the board.  I sent it to the PCB/laptop hw fix shop yesterday, and let the genius take a look at the board with microscope.  He told me that some of the tiny chips on the board nearby the flash chip was carved out badly--while I was clipping the flash chip.  (I was using a lot of force because I couldn't let the clip sit on the chip properly. That probably killed the board.)

He told me that he will 'try' to fix it (it's fix-your-laptop commercial service)  To make his life easier, I gave him identical x1c6 for his reference.
Within a week, I will get it back bricked or "fixed" (finger crossed :)   

You might want to close this issue now since it looks like my fault only, yet it might be beneficial for others to be "well prepared" when BIOS modding.    Btw,  if the brick return alive, I plan to ch134a again, with caution, to finish my job second time  :)

**Comment by tylernguyen** [comment]
@jaejunh 

Sorry about news. Hopefully the fix is easy and won't cost you much.

Definitely be more gentle with the BIOS chip and the surroundings next time!
